src/Qir/Runtime/lib/QIR/QirOutputHandling.cpp

```diff
@@ -15,9 +16,61 @@ static void PrintCStr(const char* cStr)
     WriteToCurrentStream(&msg);
 }
 
+static std::string EscapeString(std::string const& s)
+{
+    std::string out = "";
+    for (std::string::const_iterator i = s.begin(), end = s.end(); i != end; i++)
+    {
+        char c = *i;
+        if (isprint(c) && c != '\\' && c != '"')
+        {
+            out += c;
+        }
+        else
+        {
+            out += '\\';
+            switch (c)
+            {
+            case '"':
+                out += '"';
+                break;
+            case '\\':
+                out += '\\';
+                break;
+            case '\t':
+                out += 't';
+                break;
+            case '\r':
+                out += 'r';
+                break;
+            case '\n':
+                out += 'n';
+                break;
+            default:
+                char const* const hexdig = "0123456789ABCDEF";
+                out += 'x';
+                unsigned char uc = (unsigned char)c;
+                out += hexdig[uc >> 4];
+                out += hexdig[uc & 0xF];
+            }
```

Comment:
(Code Reliability)
Consider placing `break;` in the end of the `default:` block:
```suggestion
            default:
                char const* const hexdig = "0123456789ABCDEF";
                out += 'x';
                unsigned char uc = (unsigned char)c;
                out += hexdig[uc >> 4];
                out += hexdig[uc & 0xF];
                break;
            }
```
I believe the `default:` block can be placed in any part inside of the `case(..) { . . . }`, including in the beginning, i.e. between the lines 33 and 34. If someone decides to refactor the code and move the `default:` block to that place (and the block does not end with `break`) then the execution of the `default:` block will _fall through_ to line 35. Such a behavior will likely be unexpected and can result in a bug.

Reply:
done